internal/venice-common/src/main/java/com/linkedin/venice/meta/PartitionDetail.java

```diff
@@ -0,0 +1,101 @@
+package com.linkedin.venice.meta;
+
+import com.linkedin.venice.pushmonitor.PartitionStatus;
+import com.linkedin.venice.pushmonitor.ReplicaStatus;
+import com.linkedin.venice.pushmonitor.StatusSnapshot;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang.StringUtils;
+
+
+public class PartitionDetail {
+  private int partitionId;
+  private List<ReplicaDetail> replicaDetails = new ArrayList<>();
+
+  public PartitionDetail() {
+  }
+
+  public PartitionDetail(int partitionId) {
+    this.partitionId = partitionId;
+  }
+
+  public int getPartitionId() {
+    return partitionId;
+  }
+
+  public void setPartitionId(int partitionId) {
+    this.partitionId = partitionId;
+  }
+
+  public List<ReplicaDetail> getReplicaDetails() {
+    return replicaDetails;
+  }
+
+  public void setReplicaDetails(List<ReplicaDetail> replicaDetails) {
+    this.replicaDetails = replicaDetails;
+  }
+
+  void addReplicaDetails(final PartitionStatus pStatus) {
+    for (Map.Entry<String, ReplicaStatus> entry: pStatus.getReplicaStatusMap().entrySet()) {
+      String instanceId = entry.getKey();
+      StatusSnapshot startStatus = entry.getValue().findStartStatus();
+      StatusSnapshot completeStatus = entry.getValue().findCompleteStatus();
```

Comment:
The logic finds the first started status and the first completed status in the history. However, the diff between them might not be the actual replica ingestion time.

Every time when server restarts (see logic in `checkConsumptionStateWhenStart` function), it will add new started and completed statuses in to the history. But statusHistory max size is 50. If more statuses are added into history, old statuses will be deleted. As a result, the original started might be deleted, and in this case we could not find the original push time. The following is an example in ei4 hybrid-store-test-shuhliu store version 3:

```
"statusHistory": [{
		"incrementalPushVersion": "",
		"status": "END_OF_PUSH_RECEIVED",
		"time": "2023-01-20T03:13:10.852195"
	},
	{
		"incrementalPushVersion": "",
		"status": "TOPIC_SWITCH_RECEIVED",
		"time": "2023-01-20T03:13:16.329787"
	},
	{
		"incrementalPushVersion": "",
		"status": "TOPIC_SWITCH_RECEIVED",
		"time": "2023-01-20T03:13:23.965095"
	},
	{
		"incrementalPushVersion": "",
		"status": "COMPLETED",
		"time": "2023-01-26T03:37:43.545423"
	},
	{
		"incrementalPushVersion": "",
		"status": "STARTED",
		"time": "2023-01-26T05:00:26.323952"
	},
	{
		"incrementalPushVersion": "",
		"status": "COMPLETED",
		"time": "2023-01-26T05:00:30.201419"
	}
...
```

Reply:
I am not aware of this behavior and thanks for pointing it out! Thinking about a strategy to address this issue, how does the following sound?

1. organize `statusHistory` is sorted in terms of `time` in an ascending order. (we might get this property free when checking it in system, probably because the replica status is only associated with one single host and we report in the time order. Otherwise, we have to sort it). 
2. real "STARTED" and "COMPLETED" status should be the first pair in the sorted list plus another condition that there should be some other status (END_OF_PUSH_RECEIVED etc) in between.
3. If we cannot find such pair, then cannot find them.

Reply:
Sounds good. (`statusHistory` is already in time ascending order. No need to sort it again.)